File: src/main/java/com/thunderbear06/ai/task/tasks/AttackEntityTask.java
package com.thunderbear06.ai.task.tasks;

import com.thunderbear06.entity.android.AndroidEntity;
import net.minecraft.world.InteractionHand;
import net.minecraft.world.entity.LivingEntity;

public class AttackEntityTask extends MoveToEntityTask {
    private int attackCooldown;

    public AttackEntityTask(AndroidEntity android, double moveSpeed, LivingEntity entity) {
        super(android, moveSpeed, entity);
    }

    @Override
    public String getName() {
        return "attacking";
    }

    @Override
    public void tick() {
        if (attackCooldown-- > 0) return;
        if (isInRange(2.0D)) {
            attackCooldown = 10;
            android.getLookControl().setLookAt(getTarget());
            android.swing(InteractionHand.MAIN_HAND);
            android.doHurtTarget(getTarget());
        } else {
            super.tick();
        }
    }
}

File: src/main/kotlin/com/thunderbear06/ai/task/tasks/AttackEntityTask.kt
package com.thunderbear06.ai.task.tasks

import com.thunderbear06.entity.android.AndroidEntity
import net.minecraft.world.InteractionHand
import net.minecraft.world.entity.LivingEntity

class AttackEntityTask(android: AndroidEntity, moveSpeed: Double, entity: LivingEntity) : MoveToEntityTask(android, moveSpeed, entity) {
    private var attackCooldown = 0

    override val name: String = "attacking"

    override fun tick() {
        if (attackCooldown-- > 0) return
        if (isInRange(2.0)) {
            attackCooldown = 10
            android.lookControl.setLookAt(target)
            android.swing(InteractionHand.MAIN_HAND)
            android.doHurtTarget(target)
        } else {
            super.tick()
        }
    }
}
